2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/metrics/MetricsService.scala

```diff
@@ -0,0 +1,68 @@
+package com.snowplowanalytics.snowplow.collectors.scalastream.metrics
+
+import java.time.Duration
+
+import akka.http.scaladsl.model.{HttpMethod, StatusCode, Uri}
+import com.snowplowanalytics.snowplow.collectors.scalastream.metrics.PrometheusMetricsService.Metrics._
+import com.snowplowanalytics.snowplow.collectors.scalastream.metrics.PrometheusMetricsService.NanosecondsInSecond
+import com.snowplowanalytics.snowplow.collectors.scalastream.model.MetricsConfig
+import io.prometheus.client.exporter.common.TextFormat
+import io.prometheus.client.{CollectorRegistry, Counter, Gauge, Histogram}
+import org.apache.commons.io.output.StringBuilderWriter
+
+trait MetricsService {
+
+  def observeRequest(method: HttpMethod, uri: Uri, status: StatusCode, duration: Duration): Unit
+
+  def report(): String
+
+}
+
+class PrometheusMetricsService(metricsConfig: MetricsConfig) extends MetricsService {
+
+  private val registry = new CollectorRegistry
+  private val requestCounter: Counter =
+    Counter.build(HttpRequestCount, HttpRequestCountHelp).labelNames(Labels: _*).register(registry)
+  private val requestDurationHistogramBuilder = Histogram.build(HttpRequestDuration, HttpRequestDurationHelp).labelNames(Labels: _*)
+  private val requestDurationHistogram: Histogram =
+    metricsConfig.durationBuckets
+      .map(buckets => requestDurationHistogramBuilder.buckets(buckets: _*).register(registry))
+      .getOrElse(requestDurationHistogramBuilder.register(registry))
+
+  private val applicationVersionGauge = Gauge.build("service_version", "Java, scala versions and runnable name")
+    .labelNames("java_version", "scala_version", "runnable")
+    .register(registry)
+
+  applicationVersionGauge.labels(StaticMetrics.javaVersion, StaticMetrics.scalaVersion, StaticMetrics.runnableName).set(1)
```

Comment:
Isn't there a better metric type than gauge for those metrics?

Reply:
We followed advice from Brian Brazil, core maintainer of prometheus regarding the software version. Have a look here: https://www.robustperception.io/exposing-the-software-version-to-prometheus